src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,244 @@
+const markoWidgets = require('marko-widgets');
+const Expander = require('makeup-expander');
+const scrollKeyPreventer = require('makeup-prevent-scroll-keys');
+const emitAndFire = require('../../common/emit-and-fire');
+const eventUtils = require('../../common/event-utils');
+const processHtmlAttributes = require('../../common/html-attributes');
+const observer = require('../../common/property-observer');
+const template = require('./template.marko');
+
+const comboboxOptionsClass = 'combobox__options';
+const comboboxExpanderClass = 'combobox__control';
+const comboboxHostSelector = `.${comboboxExpanderClass} > input`;
+const comboboxBtnClass = 'combobox__control';
+const comboboxOptionSelector = '.combobox__option[role=option]';
+
+function getInitialState(input) {
+    const options = (input.options || []).map(option => {
+        const selected = option.selected;
+
+        return {
+            class: option.class,
+            value: option.value,
+            label: option.label,
+            selected: Boolean(selected),
+            htmlAttributes: processHtmlAttributes(option),
+            renderBody: option.renderBody
+        };
+    });
+
+    const selectedOption = options.filter(option => option.selected)[0] || options[0];
+
+    if (options.length > 0 && selectedOption.value === options[0].value) {
+        options[0].selected = true;
+    }
+
+    return {
+        name: input.name,
+        class: input.class,
+        options,
+        selected: selectedOption,
+        grow: input.grow,
+        disabled: input.disabled,
+        borderless: Boolean(input.borderless),
+        htmlAttributes: processHtmlAttributes(input)
+    };
+}
+
+function getTemplateData(state) {
+    const comboboxClass = ['combobox', state.class];
+    const btnClass = [comboboxBtnClass];
+    const optionsClass = [comboboxOptionsClass];
+
+    if (state.borderless) {
+        btnClass.push('combobox__control--borderless');
+    }
+
+    return {
+        class: comboboxClass,
+        btnClass,
+        optionsClass,
+        name: state.name,
+        selectedOption: state.selected,
+        options: state.options,
+        grow: state.grow,
+        disabled: state.disabled,
+        htmlAttributes: state.htmlAttributes
+    };
+}
+
+function init() {
+    const optionEls = this.el.querySelectorAll(comboboxOptionSelector);
+
+    if (this.state.options && this.state.options.length > 0) {
+        this.expander = new Expander(this.el, {
```

Comment:
Maybe we should change the expander to be initialized after Marko renders (onUpdate) instead of here.
I think that with this setup if you first render a select with no options, and then again with options it would not work properly.

Reply:
Also if we were to do that we could change it so that the `makeup-expander` is not initialized if the the select is disabled (or empty).

Reply:
Yes, we've found that this was an issue. This would clear that up.